src/main/java/org/radarcns/webapp/resource/DataSetEndPoint.java

```diff
@@ -158,20 +166,40 @@ public Dataset getSamplesJson(
         // todo: 404 if given source does not exist.
         // Note that a source doesn't necessarily need to be linked anymore, as long as it exists
         // and historical data of it is linked to the given user.
-        mpClient.getSubject(subjectId);
+        mpClient.checkSubjectInProject(projectName, subjectId);
         Dataset dataset;
 
-        TimeWindow timeWindow = interval != null ? interval : TEN_SECOND;
+        // Don't request future data
+        Instant endTime = end.getValue();
+        if (endTime == null) {
+            endTime = Instant.now();
+        }
 
-        if (start != null && end != null) {
-            dataset = dataSetService.getAllRecordsInWindow(projectName,
-                    subjectId, sourceId, sourceDataName, stat, timeWindow,
-                    Date.from(start.getValue()), Date.from(end.getValue()));
-        } else {
-            dataset = dataSetService.getAllDataItems(projectName, subjectId,
-                    sourceId, sourceDataName, stat, timeWindow);
+        TimeWindow timeWindow = interval;
+        Instant startTime = start.getValue();
+        TimeFrame timeFrame = new TimeFrame(startTime, endTime);
+
+        if (startTime != null && startTime.isAfter(endTime)) {
+            // don't mix up time frame order
+            timeFrame.setStartDateTime(endTime);
+        } else if (startTime == null && timeWindow == null) {
+            // default settings, 1 year with 1 week intervals
+            timeWindow = TEN_SECOND;
```

Comment:
Should be `ONE_WEEK` instead of `TEN_SECOND`. Perhaps fix this at the source as well.